libtenzir/include/tenzir/select_optimization.hpp

```diff
@@ -0,0 +1,41 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <caf/default_sum_type_access.hpp>
+#include <caf/detail/type_list.hpp>
+#include <caf/none.hpp>
+#include <caf/variant.hpp>
+
+#include <string>
+#include <vector>
+
+namespace tenzir {
+
+class select_optimization {
+public:
+  std::vector<std::string> fields_of_interest{};
+  select_optimization() = default;
+  explicit select_optimization(std::vector<std::string> fields_of_interest)
+    : fields_of_interest{std::move(fields_of_interest)} {
+  }
+
+  static auto no_select_optimization() -> select_optimization {
+    return select_optimization(std::vector<std::string>{});
+  }
+};
+
+template <class Inspector>
+auto inspect(Inspector& f, select_optimization& x) {
+  return f.object(x)
+    .pretty_name("expression")
+    .fields(f.field("node", x.fields_of_interest));
+}
```

Comment:
Please make this a friend function of the struct to make it a bit more ADL-friendly.

```suggestion
friend auto inspect(auto& f, select_optimization& x) -> bool {
  return f.object(x)
    .fields(f.field("fields", x.fields_of_interest));
}
```